From 01a582e6edee238d995c8514c4bd1b90a1902897 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Fri, 6 Jan 2017 13:53:13 +0700
Subject: [PATCH 14/28] 
 system-core_WW04-Re-introduce-lpm_boot-for-offline-charging.patch

Change-Id: Ie424140c18dbfe5b59ecc4aa24e9a871b3b48aee
---
 init/Android.mk |  5 ++++-
 init/init.cpp   | 15 ++++++++++++++-
 2 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/init/Android.mk b/init/Android.mk
index 0189ecd..4942468 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -78,7 +78,10 @@ LOCAL_SRC_FILES:= \
 
 SYSTEM_CORE_INIT_DEFINES := BOARD_CHARGING_MODE_BOOTING_LPM \
     BOARD_CHARGING_CMDLINE_NAME \
-    BOARD_CHARGING_CMDLINE_VALUE
+    BOARD_CHARGING_CMDLINE_VALUE \
+    BOARD_LPM_BOOT_ARGUMENT_NAME \
+    BOARD_LPM_BOOT_ARGUMENT_VALUE
+
 
 $(foreach system_core_init_define,$(SYSTEM_CORE_INIT_DEFINES), \
   $(if $($(system_core_init_define)), \
diff --git a/init/init.cpp b/init/init.cpp
index 34dcc37..007be23 100755
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -78,6 +78,11 @@ static int property_triggers_enabled = 0;
 #define BOARD_CHARGING_CMDLINE_VALUE "true"
 #endif
 
+#ifndef BOARD_LPM_BOOT_ARGUMENT_NAME
+#define BOARD_LPM_BOOT_ARGUMENT_NAME "lpm_boot"
+#define BOARD_LPM_BOOT_ARGUMENT_VALUE "1"
+#endif
+
 static char qemu[32];
 static char battchg_pause[32];
 
@@ -87,6 +92,8 @@ static time_t process_needs_restart;
 
 const char *ENV[32];
 
+static unsigned lpm_bootmode = 0;
+
 bool waiting_for_exec = false;
 
 static int epoll_fd = -1;
@@ -334,6 +341,12 @@ static void import_kernel_nv(const std::string& key, const std::string& value, b
 
     if (key == "qemu") {
         strlcpy(qemu, value.c_str(), sizeof(qemu));
+#ifdef BOARD_LPM_BOOT_ARGUMENT_NAME
+    } else if (key == BOARD_LPM_BOOT_ARGUMENT_NAME) {
+        if (!strcmp(value.c_str(), BOARD_LPM_BOOT_ARGUMENT_VALUE)) {
+            lpm_bootmode = 1;
+        }
+#endif
     } else if (key == BOARD_CHARGING_CMDLINE_NAME) {
         strlcpy(battchg_pause, value.c_str(), sizeof(battchg_pause));
     } else if (android::base::StartsWith(key, "androidboot.")) {
@@ -533,7 +546,7 @@ static void selinux_initialize(bool in_kernel_domain) {
 
 static int charging_mode_booting(void) {
 #ifndef BOARD_CHARGING_MODE_BOOTING_LPM
-    return 0;
+    return lpm_bootmode;
 #else
     int f;
     char cmb;
-- 
2.9.3

